nixos/modules/services/misc/matrix-dendrite.nix

```diff
@@ -0,0 +1,179 @@
+              The domain name of the server, with optional explicit port.
+              This is used by remote servers to connect to this server,
+              e.g. matrix.org, localhost:8080, etc.
+              This is also the last part of your UserID.
+            '';
+          };
+          private_key = lib.mkOption {
+            type = lib.types.nullOr lib.types.path;
+            example = "${workingDir}/matrix_key.pem";
+            default = "${workingDir}/matrix_key.pem";
+            description = ''
+              The path to the signing private key file, used to sign
+              requests and events.
+            '';
+          };
+          trusted_third_party_id_servers = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            example = [ "matrix.org" ];
+            default = [ "matrix.org" "vector.im" ];
+            description = ''
+              Lists of domains that the server will trust as identity
+              servers to verify third
+            '';
+          };
+          disable_federation = lib.mkOption {
+            type = lib.types.bool;
+            default = true;
+            description = ''
+              Whether to enable federation with other matrix servers.
+            '';
+          };
+        };
+        options.client_api = {
+          registration_disabled = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              Whether to disable user registration to the server
+              without the shared secret.
+            '';
+          };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for dendrite, see:
+        <link xlink:href="https://github.com/matrix-org/dendrite"/>
+      '';
+    };
+  };
+
+  config = lib.mkMerge [
+    (lib.mkIf cfg.enable {
+      systemd.services.matrix-dendrite = {
+        description = "Dendrite Matrix homeserver";
+        after = [
+          "network.target"
+        ] ++ lib.optionals (cfg.generate_private_key == true) [
+          "matrix-dendrite-key-generator.service"
+        ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Type = "notify";
+          DynamicUser = true;
+          StateDirectory = cfg.stateDir;
+          WorkingDirectory = workingDir;
+          ExecStart = lib.strings.concatStringsSep " " ([
+            "${pkgs.matrix-dendrite}/bin/dendrite-monolith-server"
+            "--config ${configurationYaml}"
+            "--http-bind-address :${builtins.toString cfg.httpPort}"
+          ] ++ lib.optionals (cfg.httpsPort != null) [
+            "--https-bind-address :${builtins.toString cfg.httpsPort}"
+          ] ++ lib.optionals (cfg.tls_cert != null) [
+            "--tls-cert ${cfg.tls_cert}"
+          ] ++ lib.optionals (cfg.tls_key != null) [
+            "--tls-key ${cfg.tls_key}"
+          ]);
+          ExecReload = "${pkgs.util-linux}/bin/kill -HUP $MAINPID";
```

Comment:
`kill` should be in `PATH`, so you can omit the absolute path of it

Reply:
This doesn't work for me.

```
matrix-dendrite.service: Failed to locate executable kill: No such file or directory
matrix-dendrite.service: Failed at step EXEC spawning kill: No such file or directory
```

Reply:
@szlend You mean as it does now? Or omitting the absolute path?

Reply:
Yeah, I'm saying currently (without the full path) reload doesn't work for me.

Reply:
So `${pkgs.util-linux}/bin/kill` doesn't work, but does `kill` by itself?

Reply:
(sorry ninja edit)

Reply:
I didn't try to fix it, but `kill` by itself is definitely not enough in this instance.

Reply:
Changed it to use pkgs.coreutils like other modules

Reply:
Good to know! Will remember this :)